From 01c446e675c106c853cfabf5470bed4a7cc6ae4b Mon Sep 17 00:00:00 2001
From: Ronny Lorenz <ronny@tbi.univie.ac.at>
Date: Tue, 8 Dec 2015 15:20:24 +0100
Subject: [PATCH 1/8] mwifiex: do not create AP and P2P interfaces upon driver
 loading

From 648705896f1b7621b0acba6ca760e553a548ca59 Mon Sep 17 00:00:00 2001
From: Bing Zhao <bzhao@marvell.com>
Date: Mon, 19 Aug 2013 16:10:21 -0700
Subject: [PATCH 1/3] mwifiex: do not create AP and P2P interfaces upon driver
 loading

Bug 60747 - 1286:2044 [Microsoft Surface Pro]
    Marvell 88W8797 wifi show 3 interface under network
https://bugzilla.kernel.org/show_bug.cgi?id=60747

This issue was also reported previously by OLPC and some folks from
the community.

There are 3 network interfaces with different types being created
when mwifiex driver is loaded:

1. mlan0 (infra. STA)
2. uap0 (AP)
3. p2p0 (P2P_CLIENT)

The Network Manager attempts to use all 3 interfaces above without
filtering the managed interface type. As the result, 3 identical
interfaces are displayed under network manager. If user happens to
click on an entry under which its interface is uap0 or p2p0, the
association will fail.

Work around it by removing the creation of AP and P2P interfaces
at driver loading time. These interfaces can be added with 'iw' or
other applications manually when they are needed.

Signed-off-by: Bing Zhao <bzhao@marvell.com>
Signed-off-by: Avinash Patil <patila@marvell.com>
---
 drivers/net/wireless-3.8/mwifiex/main.c | 15 ---------------
 1 file changed, 15 deletions(-)

diff --git a/drivers/net/wireless-3.8/mwifiex/main.c b/drivers/net/wireless-3.8/mwifiex/main.c
index 17b7c05d..3804ed9 100644
--- a/drivers/net/wireless-3.8/mwifiex/main.c
+++ b/drivers/net/wireless-3.8/mwifiex/main.c
@@ -523,21 +523,6 @@ static void mwifiex_fw_dpc(const struct firmware *firmware, void *context)
 		goto err_add_intf;
 	}
 
-	/* Create AP interface by default */
-	if (!mwifiex_add_virtual_intf(adapter->wiphy, "uap%d",
-				      NL80211_IFTYPE_AP, NULL, NULL)) {
-		mwifiex_dbg(adapter, ERROR,
-			    "cannot create default AP interface\n");
-		goto err_add_intf;
-	}
-
-	/* Create P2P interface by default */
-	if (!mwifiex_add_virtual_intf(adapter->wiphy, "p2p%d",
-				      NL80211_IFTYPE_P2P_CLIENT, NULL, NULL)) {
-		mwifiex_dbg(adapter, ERROR,
-			    "cannot create default P2P interface\n");
-		goto err_add_intf;
-	}
 	rtnl_unlock();
 
 	mwifiex_drv_get_driver_version(adapter, fmt, sizeof(fmt) - 1);
-- 
2.6.3

